src/core/surface/server_create.c

```diff
@@ -43,9 +43,6 @@ grpc_server *grpc_server_create(const grpc_channel_args *args, void *reserved) {
   const grpc_channel_filter *filters[3];
   size_t num_filters = 0;
   filters[num_filters++] = &grpc_compress_filter;
-  if (grpc_channel_args_is_census_enabled(args)) {
-    filters[num_filters++] = &grpc_server_census_filter;
-  }
```

Comment:
In the grpc_server_create_from_filters which is in the server.c . You commented that line then ctiller added this and we end up with 2 initialization logic but one was never gonna work.